standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/RawStore.java

```diff
@@ -459,16 +459,15 @@ boolean doesPartitionExist(String catName, String dbName, String tableName,
    * @param catName catalog name.
    * @param dbName database name.
    * @param tableName table name.
-   * @param part_vals list of partition values.
+   * @param partName partition name.
    * @return true if the partition was dropped.
    * @throws MetaException Error accessing the RDBMS.
    * @throws NoSuchObjectException no partition matching this description exists
    * @throws InvalidObjectException error dropping the statistics for the partition
    * @throws InvalidInputException error dropping the statistics for the partition
    */
-  boolean dropPartition(String catName, String dbName, String tableName,
-      List<String> part_vals) throws MetaException, NoSuchObjectException, InvalidObjectException,
-      InvalidInputException;
+  boolean dropPartition(String catName, String dbName, String tableName, String partName)
```

Comment:
That is a change in API that breaks backward compatibility.

Reply:
Make sense, my initial thought was that this is an internal API which is not exposed to clients, so compatibility was not considered. I will follow your comment.

Reply:
IMHO, Instead of defining this new API, we can make the partname inside the existing `dropPartition()` method itself to invoke `dropPartitionsInternal()` or `dropPartitions()`. Because the new API signature is similar to exisiting `dropPartitions()` except the last argument(i.e., single partname vs list of partnames). May be, we can mark `dropPartition()` as deprecated and insist using `dropPartitions()` directly in future.

Reply:
If we reuse the existing `dropPartition()` method, we need to get the `table` and partition keys again in `ObjectStore` to make the partname, so I think it's reasonable to create this new API and deprecate the old one.

Reply:
I would favor this change instead of passing the table into the drop partition API in the object store.

Reply:
I'm Ok with your current changes(a new api in the raw store).

Reply:
Got it! Thank you.